pilot/model/config.go

```diff
@@ -530,3 +576,117 @@ func RejectConflictingEgressRules(egressRules map[string]*proxyconfig.EgressRule
 
 	return filteredEgressRules, errs
 }
+
+// `istio.mixer.v1.config.client.IstioService` and
+// `istio.proxy.v1.config.IstioService` are logically
+// equivalent. Convert from mixer-to-proxy representation so we can
+// use ResolveHostname below.
```

Comment:
note: `IstioService` is going away in pilot (see @louiscryan and @rshriram latest gateway proposal)